libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -553,22 +453,16 @@ void UARTDriver::_timer_tick(void)
     }
 
     // try to fill the read buffer
-    uint16_t _head;
-    n = BUF_SPACE(_readbuf);
-    if (n > 0) {
-        uint16_t n1 = _readbuf_size - _readbuf_tail;
-        if (n1 >= n) {
-            // one read will do
-            assert(_readbuf_tail+n <= _readbuf_size);
-            _read_fd(&_readbuf[_readbuf_tail], n);
-        } else {
-            assert(_readbuf_tail+n1 <= _readbuf_size);
-            int ret = _read_fd(&_readbuf[_readbuf_tail], n1);
-            if (ret == n1 && n > n1) {
-                assert(_readbuf_tail+(n-n1) <= _readbuf_size);
-                _read_fd(&_readbuf[_readbuf_tail], n - n1);
-            }
-        }
+    int ret;
+    ByteBuffer::IoVec vec[2];
+
+    const auto n_vec = _readbuf.reserve(vec, _readbuf.space());
```

Comment:
That's my point, auto is never necessary and should be removed.

On Jul 30, 2016 8:13 AM, "Murilo Belluzzo" <notifications@github.com> wrote:

> In libraries/AP_HAL_Linux/UARTDriver.cpp
> <https://github.com/ArduPilot/ardupilot/pull/4575#discussion_r72889080>:
>
> > -        if (n1 >= n) {
> > -            // one read will do
> > -            assert(_readbuf_tail+n <= _readbuf_size);
> > -            _read_fd(&_readbuf[_readbuf_tail], n);
> > -        } else {
> > -            assert(_readbuf_tail+n1 <= _readbuf_size);
> > -            int ret = _read_fd(&_readbuf[_readbuf_tail], n1);
> > -            if (ret == n1 && n > n1) {
> > -                assert(_readbuf_tail+(n-n1) <= _readbuf_size);
> > -                _read_fd(&_readbuf[_readbuf_tail], n - n1);
> > -            }
> > -        }
> > +    int ret;
> > +    ByteBuffer::IoVec vec[2];
> > +
> > +    const auto n_vec = _readbuf.reserve(vec, _readbuf.space());
>
> Is any auto necessary? It is just a language perk, no performance
> penalties.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ArduPilot/ardupilot/pull/4575/files/cdea244c4fa69585c154746f96608bcf3434af5f#r72889080>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEj7G753QTfyyfNLqPv3HDrkh7imCtEGks5qa2oBgaJpZM4JYzVo>
> .
>